### ISSUE TITLE
rAdam may be wrongly implemented

### ISSUE THREAD

**Issue by chenweize1998** [geoopt/geoopt#164]
I'm not so familiar with optimizers, but when I checked the code of rAdam, I found something unnatural. I think it is very likely to be a bug and somebody who knows well about adam and Riemann optimization should look into it.
https://github.com/geoopt/geoopt/blob/244bcd0ed85eb7c32ebfec96565a99f1c4c3680f/geoopt/optim/radam.py#L104-L122

The `group['step']` is added twice in line 104 and line 122, which seems incorrect, and neither line seems to be the right place to update `group['step']`.

Compared with Adam implemented by PyTorch official, the `group['step'] += 1` should only appear inside the loop over `param_groups`, outside the loop over `group['params']`. I think it should be updated in the following code block.
https://github.com/geoopt/geoopt/blob/244bcd0ed85eb7c32ebfec96565a99f1c4c3680f/geoopt/optim/radam.py#L52-L59

**Comment by ferrine** [comment]
That is a shocking typo. Thanks for pointing that out!!!!

**Comment by chenweize1998** [comment]
Moreover, I found that `eps` will be scaled, which is not expected.
https://github.com/geoopt/geoopt/blob/0f3adeb952315ce6a63f70c60998c28081a119fd/geoopt/optim/radam.py#L103
The `eps` is added to `exp_avg_sq` (moving variance before debiasing), and the debias factor `1/(1-\beta^t)` for both moving average and moving variance are both combined with learning rate in the following manner
https://github.com/geoopt/geoopt/blob/0f3adeb952315ce6a63f70c60998c28081a119fd/geoopt/optim/radam.py#L106-L112
At last, `-step_size * direction` will actually be `\eta\frac{m_t/(1-\beta_1^t)}{\sqrt{v_t}/\sqrt{(1-\beta_2^t)}+\epsilon/\sqrt{(1-\beta_2^t)}/}`, that is, `eps` will also be scaled up by `1/(1-\beta_2^t)`. When `beta_2=0.999`, the `eps` will even be scaled up by 31x when training just starts.

**Comment by ferrine** [comment]
Hmm
